staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -224,6 +224,12 @@ func TestWatchFromNoneZero(t *testing.T) {
 }
 
 func TestWatchError(t *testing.T) {
+	// we need to check for watch errors so do not panic the server
```

Comment:
@deads2k @wojtek-t so the `panic` from the decode error happens in a different go routine and thus I do not think the test can directly `recover` from it.  I added some code that uses a `defer` in the non-test code, but it all seems a bit hacky to me.